### PR TITLE
Change testing import strategy

### LINKED ISSUE #413: Change test import logic to make local testing easier
@erogluorhan, tagging you as a follow up from our conversation about this last week

Currently, we do imports in our testing files generally like this:
``` python
# Import from directory structure if coverage test, or from installed
# packages otherwise
if "--cov" in str(sys.argv):
    from src.geocat.comp import (whatever)
else:
    from geocat.comp import (whatever)
```

This looks for some sort of install of `geocat.comp` in the environment to run the tests against instead of importing from the file structure, unless running coverage tests. This is clunky for local testing. 

I propose we do something more like the following:
``` python
# Import from directory structure unless running tests on gh actions
if os.environ.get('GITHUB_ACTIONS') == 'true':
    from geocat.comp import (whatever)
else:
    from src.geocat.comp import (whatever)
```

This will allow for import from the file structure in all cases except for when the github actions tests are being run. This will allow for more intuitive local testing while leaving the behaviour for coverage tests and github action tests unchanged.

### PR DESCRIPTION
**Pull request by anissa111**
## PR Summary
Closes #413 

This PR changes the import strategy for our testing to match that of uxarray's, where the tests always import from the directory structure instead of an installed package.

## PR Checklist

**General**
- [x] Make an issue if one doesn't already exist
- [x] Link the issue this PR resolves by adding `closes #XXX` to the PR description where XXX is the number of the issue.
- [x] Add a brief summary of changes to `docs/release-notes.rst`
- [x] Add appropriate labels to this PR
- [x] Make your changes in a forked repository rather than directly in this repo
- [x] Passes `precommit`. To set up on your local, run `pre-commit install` from the top level of the repository. To manually run pre-commits, use `pre-commit run --all-files` and re-add any changed files before committing again and pushing.

### CODE REVIEW

**Review by kafitzgerald — APPROVED** [review]
Looks good to me.  

That said, I haven't been around long enough to know if anyone might have other perspectives that should be considered.

**Comment by erogluorhan** [comment]
> Yeah! I'd really like to get @erogluorhan's look at this before I merge if possible

Thank you! Please see my [comment](https://github.com/NCAR/geocat-comp/issues/413#issuecomment-1559731653) in the linked issue.

File: test/test_climatologies.py [hunk 1]
```diff
@@ -7,12 +6,14 @@
 from parameterized import parameterized
 import xarray as xr
 
-# Import from directory structure if coverage test, or from installed
-# packages otherwise
-if "--cov" in str(sys.argv):
-    from src.geocat.comp import climate_anomaly, anomaly, climatology, month_to_season, calendar_average, climatology_average
-else:
-    from geocat.comp import climate_anomaly, anomaly, climatology, month_to_season, calendar_average, climatology_average
+from src.geocat.comp import (
```
Comment (erogluorhan) [inline test/test_climatologies.py:9]: With this change, we are opting out from testing against the install directory and sticking only to reading from the source directory. I believe the former would be useful to catch in CI tests potential, rare issues with package installation.

**Comment by anissa111** [comment]
Closing in favor of a little package reorganization as discussed with @erogluorhan

We're going to try removing the top level `src/` folder and just having the nested `geocat/comp/` in the top level instead.

**Pull request closed without merging**